src/libs/nav.ts

```diff
@@ -128,3 +160,13 @@ export function PathHashInserter() {
   });
   return null;
 }
+
+export interface TerraNavLinkProvider {
+  getLink: (name: string, params?: Record<string, any>) => string;
+  goToPath: (name: string, params?: Record<string, any>) => void;
```

Comment:
I am confused as to why these names are different. I may need some in-person help to grok this.

Reply:
as we discussed, this is a mapping interface to help the current nav implementation in terra-ui be mappable to the new more generic NavLinkProvider interface, but still honor the existing method names.  We may iterate later and add more TS goodness, but this is meant as a babystep.